### PR TITLE
WIP Add pam_tty_audit.so to sudo, login pam services

### PR DESCRIPTION
**Pull request by mkg20001**
Fixes PL-129625

~~Currently it doesn't seem to log anything and `aureport --tty ` is only complaining about missing files~~

~~I suspect something needs to be enabled for this to work?~~

@flyingcircusio/release-managers

## Release process

Impact:

Changelog:

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

### CODE REVIEW

File: nixos/infrastructure/testing.nix [hunk 1]
```diff
@@ -18,11 +18,19 @@
     services.openssh.enable = lib.mkOverride 60 false;
     # build-vms.nix from NixOS automatically generates numbered interface
     # configs with default IPs. We rename the devices to fe and srv early so
-    # the services wait for eth1 and eth2 for 5 minutes and time out.
+    # the services wait for the interfaces for 5 minutes and time out.
```
Comment (ctheune) [inline nixos/infrastructure/testing.nix:21]: I like this! Can we extract that into a separate PR, please?

File: versions.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "nixpkgs": {
-    "owner": "NixOS",
+    "owner": "mkg20001",
```
Comment (ctheune) [inline versions.json]: We need to clean this up before merging!

File: tests/testlib.nix [hunk 3]
```diff
@@ -1,5 +1,26 @@
 { lib }:
-{
+
```
Comment (ctheune) [inline tests/testlib.nix:2]: This sounds good, I'd like if we could merge that together with the separate PR from the first change in here.

**Review by ctheune — CHANGES_REQUESTED** [review]

File: versions.json (hunk 2 above)
Comment (mkg20001) [reply versions.json]: upstream PR for this is https://github.com/NixOS/nixpkgs/pull/137661

File: nixos/infrastructure/testing.nix (hunk 1 above)
Comment (mkg20001) [reply nixos/infrastructure/testing.nix:21]: Has been split https://github.com/flyingcircusio/fc-nixos/pull/372

Comment (mkg20001) [reply nixos/infrastructure/testing.nix:21]: Note that I'd need that other PR merged before this one then

Comment (ctheune) [reply nixos/infrastructure/testing.nix:21]: Done

**Comment by ctheune** [comment]
I'm merging this onto a branch to make a few adjustments.
